### PR TITLE
23 data ingestion

### LINKED ISSUE #23: Create Data Ingestion Service
*Acceptance criteria*

- Data is POSTed to `/ingest` endpoint
- Data is stored in Mongo

*Technical notes*

- Use mongodb
- Use PyMongo

### PR DESCRIPTION
**Pull request by ghostaccount737**
Closes #23

### CODE REVIEW

File: inference/app/__init__.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import os
 from flask import Flask, render_template, request, jsonify, make_response
+from flask_pymongo import PyMongo
 import werkzeug
 from werkzeug.utils import secure_filename
 
 app = Flask(__name__, instance_relative_config=True)
 app.config.from_object('config.TestingConfig')
+mongo = PyMongo(app, uri="mongodb://localhost:27017/customerapp")
```
Comment (ghostaccount737) [inline inference/app/__init__.py]: Need modifications here

**Review by SapoonDutta — APPROVED** [review]

File: inference/tests/test_app.py [hunk 2]
```diff
@@ -5,4 +5,6 @@
 def test_uploadfile():
     client = app.test_client()
     response = client.post("/upload_file", data=json.dumps(dict(f='f')))
-    assert response.status_code == 200
+    assert response.status_code == 400
```
Comment (lastlegion) [inline inference/tests/test_app.py:8]: why remove the success 200 case?

File: inference/app/__init__.py [hunk 3]
```diff
@@ -2,9 +2,16 @@
...
+import pymongo
+import json
 
 app = Flask(__name__, instance_relative_config=True)
 app.config.from_object('config.TestingConfig')
+URI = 'mongodb://127.0.0.1:27017'
+client = pymongo.MongoClient(URI)
+db = client['test']
+vizrec = db.vizrec
+print(vizrec)
```
Comment (lastlegion) [inline inference/app/__init__.py]: remove

File: inference/app/__init__.py [hunk 4]
```diff
@@ -2,9 +2,16 @@
 from flask import Flask, render_template, request, jsonify, make_response
 import werkzeug
 from werkzeug.utils import secure_filename
+import pymongo
+import json
 
 app = Flask(__name__, instance_relative_config=True)
 app.config.from_object('config.TestingConfig')
+URI = 'mongodb://127.0.0.1:27017'
```
Comment (lastlegion) [inline inference/app/__init__.py]: make this configurable

File: inference/app/__init__.py [hunk 5]
```diff
@@ -2,9 +2,16 @@
...
+client = pymongo.MongoClient(URI)
+db = client['test']
```
Comment (lastlegion) [inline inference/app/__init__.py]: make database host, port, dbname, collection-name configurable

File: inference/tests/test_app.py (hunk 2 above)
Comment (ghostaccount737) [reply inference/tests/test_app.py:8]: We are giving .json files as input

Comment (lastlegion) [reply inference/tests/test_app.py:8]: please create a success test case and a failure one.

Comment (ghostaccount737) [reply inference/tests/test_app.py:8]: The addition can soon be found in #35

**Pull request merged**